src/pages/SignUp/SignUp.jsx

```diff
@@ -0,0 +1,175 @@
+import React from 'react';
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+import TermsDialog from './TermsDialog';
+import axios from '../../utils/axios';
+import { useNavigate } from 'react-router-dom';
+import Box from '@mui/material/Box';
+import {
+  Avatar,
+  Button,
+  Checkbox,
+  Container,
+  FormControlLabel,
+  Grid,
+  Link,
+  TextField,
+  Typography,
+} from '@mui/material';
+import { useTranslation } from 'react-i18next';
+
+export default function SignUp() {
+  const [open, setOpen] = React.useState(false);
+  const [isAgree, setIsAgree] = React.useState(false);
```

Comment:
We already have this state in formik(acceptTerms), we don't need to use two states for the same thing. 😉
`isAgree` it can become `formik.values.acceptTerms`
`setIsAgree` it can become `formik.setFieldValue`

So `TermsDialog` will become:
`<TermsDialog open={open} setOpen={setOpen} setFieldValue={formik.setFieldValue} />`